File: src/main/scala/com/socrata/api/SchemaApi.java
package com.socrata.api;

import com.socrata.models.SchemaRequest;
import com.socrata.models.SchemaResponse;
import io.smallrye.mutiny.Uni;
import jakarta.ws.rs.BeanParam;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

@Path("/schema")
public interface SchemaApi {

    @GET
    @Produces({MediaType.APPLICATION_JSON})
    Uni<SchemaResponse> getSchema(@BeanParam SchemaRequest schemaRequest);
}

File: src/main/scala/com/socrata/api/SchemaApi.scala
package com.socrata.api

import com.socrata.datacoordinator.truth.metadata.SchemaWithFieldName
import io.smallrye.mutiny.Uni
import jakarta.ws.rs._
import jakarta.ws.rs.core.MediaType
import org.jboss.resteasy.reactive.RestQuery

import java.util.Optional
import scala.annotation.meta.field


case class SchemaRequest() {
  @(RestQuery@field) var copy: String = null
  @(RestQuery@field) var fieldName: Optional[java.lang.Boolean] = Optional.empty()
  @(RestQuery@field) var ds: Optional[String] = Optional.empty()
  @(RestQuery@field) var rn: Optional[String] = Optional.empty()
}


@Path("/schema")
trait SchemaApi {
  @GET
  @Produces(Array[String](MediaType.APPLICATION_JSON))
  def getSchema(@BeanParam schemaRequest: SchemaRequest): Uni[SchemaWithFieldName]
}
